SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestCharacterMovement/SpatialTestCharacterMovement.cpp

```diff
@@ -111,28 +98,38 @@ void ASpatialTestCharacterMovement::PrepareTest()
 			return IsValid(PlayerCharacter) && FMath::IsNearlyEqual(PlayerCharacter->GetActorLocation().Z, 40.0f, 2.0f);
 		},
 		nullptr,
-		[this](float DeltaTime) {
+		[this, DestinationPlane](float DeltaTime) {
 			AController* PlayerController = Cast<AController>(GetLocalFlowController()->GetOwner());
 			ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
 
 			PlayerCharacter->AddMovementInput(FVector(1, 0, 0), 1.0f);
 
-			if (bCharacterReachedDestination)
-			{
-				AssertTrue(bCharacterReachedDestination, TEXT("Player character has reached the destination on the autonomous proxy."));
-				FinishStep();
-			}
+			RequireTrue(HasCharacterReachedDestination(PlayerCharacter, DestinationPlane),
```

Comment:
We haven't allowed any time for the movement to happen, should some of these DestinationPlanes be flipped across the X axis?

Reply:
It's my understanding that the RequireTrue allows for failure and will prevent the FinishStep from moving on until this is true. So it allows forward movement until the character reaches the plane.

Reply:
In this test, we only move forward so I don't think we need to flip the plane.

Reply:
Ah yeah you're right - otherwise the step times out for failure. Makes sense!